src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -185,21 +188,58 @@ public TripMonitorNotification checkTripForNewAlerts() {
         return notification;
     }
 
-    public TripMonitorNotification checkTripForDepartureDelay() {
-        long departureDelayInMinutes = TimeUnit.SECONDS.toMinutes(matchingItinerary.legs.get(0).departureDelay);
-        // First leg departure time should not exceed variance allowed.
-        if (departureDelayInMinutes >= trip.departureVarianceMinutesThreshold) {
-            return TripMonitorNotification.createDelayNotification(departureDelayInMinutes, trip.departureVarianceMinutesThreshold, DEPARTURE_DELAY);
+    /**
+     * Checks whether the trip is beginning or ending at a time greater than the allowable variance. This will check
+     * whether the departure or arrival time of the whole journey has deviated to the point where the absolute value of
+     * the variance has changed more than the trip's variance. If it has, a notification is generated.
+     */
+    public TripMonitorNotification checkTripForDelay(NotificationType delayType) {
+        // the target time for trip depending on notification type
+        Date matchingItineraryTargetTime;
+
+        // the current baseline epoch millis to check if a new threshold has been met
+        long baselineItineraryTargetEpochMillis;
+
+        // the original target epoch millis that the trip would've started or ended at
+        long originalTargetTimeEpochMillis;
+
+        if (delayType == NotificationType.DEPARTURE_DELAY) {
+            matchingItineraryTargetTime = matchingItinerary.startTime;
+            baselineItineraryTargetEpochMillis = journeyState.baselineDepartureTimeEpochMillis;
+            originalTargetTimeEpochMillis = journeyState.originalDepartureTimeEpochMillis;
+        } else {
+            matchingItineraryTargetTime = matchingItinerary.endTime;
+            baselineItineraryTargetEpochMillis = journeyState.baselineArrivalTimeEpochMillis;
+            originalTargetTimeEpochMillis = journeyState.originalArrivalTimeEpochMillis;
         }
-        return null;
-    }
 
-    public TripMonitorNotification checkTripForArrivalDelay() {
-        Leg lastLeg = matchingItinerary.legs.get(matchingItinerary.legs.size() - 1);
-        long arrivalDelayInMinutes = TimeUnit.SECONDS.toMinutes(lastLeg.arrivalDelay);
-        // Last leg arrival time should not exceed variance allowed.
-        if (arrivalDelayInMinutes >= trip.arrivalVarianceMinutesThreshold) {
-            return TripMonitorNotification.createDelayNotification(arrivalDelayInMinutes, trip.arrivalVarianceMinutesThreshold, ARRIVAL_DELAY);
+        // calculate deviation from threshold in minutes
```

Comment:
Is this the deviation from threshold or the deviation from the scheduled arrival/departure?

Reply:
Clarified comment in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14